app/javascript/controllers/scroll_controller.js

```diff
@@ -5,6 +5,18 @@ export default class extends Controller {
     static targets = ["listItem", "scrollItem"]
     currentSelectedIndex = 0;
 
+    connect() {
+        document.addEventListener("scroll", () => {
+            this.scrollEvent();
+        });
+    }
+
+    disconnect() {
```

Comment:
Why is that needed?

Reply:
So it does not trigger the event listener when scrolling anywhere else on the page.